File: 2256-minimum-average-difference/2256-minimum-average-difference.go
func minimumAverageDifference(nums []int) int {
    var sum int64 = 0
    for _, num := range(nums) {
        sum += int64(num)
    }
    n := len(nums)
    ans := 0
    minAvg := math.MaxInt
    var currSum int64 = int64(nums[0])
    for i := 1; i < n; i++ {
        avg := int((currSum / int64(i)) - (sum - currSum) / int64(n - i));
        avg = int(math.Abs(float64(avg)))
        currSum += int64(nums[i]);
        if(avg < minAvg) {
            ans = i - 1;
            minAvg = avg;
        }
    }
    if(int(sum / int64(n)) < minAvg) {
        ans = n - 1;
    }
    return ans;
}

File: 2256-minimum-average-difference/2256-minimum-average-difference.cpp
class Solution {
public:
    int minimumAverageDifference(vector<int>& nums) {
        long long sum = accumulate(nums.begin(), nums.end(), 0L);
        int n = nums.size();
        int avg;
        int ans = 0;
        int minAvg = INT_MAX;
        long long currSum = nums[0];
        for(int i = 1; i < n; i++) {
            avg = (currSum / i) - (sum - currSum) / (n - i);
            avg = abs(avg);
            currSum += nums[i];
            if(avg < minAvg) {
                ans = i - 1;
                minAvg = avg;
            }
        }
        if(sum / n < minAvg) ans = n - 1;
        return ans;
    }
};